src/framework/auth/services/auth.service.ts

```diff
@@ -34,12 +47,29 @@ export class NbAuthService {
   }
 
   /**
-   * Returns true if auth token is presented in the token storage
+   * Returns true if valid auth token is present in the token storage.
+   * If not, calls the strategy refreshToken, and returns true if it get a new valid access token, false otherwise
    * @returns {Observable<any>}
    */
   isAuthenticated(): Observable<boolean> {
```

Comment:
I guess this first of all should be under a parameter `refresh = false` or even a separate method, like `isAuthenticatedOrRefresh()`.

Reply:
Yes it will be clearer .